src/udx.c

```diff
@@ -1096,6 +1096,11 @@ udx__unshift_packet (udx_packet_t *pkt, udx_socket_t *socket) {
 
   if (pkt->type == UDX_PACKET_TYPE_STREAM_DESTROY) {
     udx_stream_t *stream = pkt->ctx;
+
+    if (pkt->seq + 1 == stream->seq) {
```

Comment:
be nice to have a comment above this if, as decrementing seqs is normally super dangerous